src/axom/primal/geometry/BoundingBox.hpp

```diff
@@ -432,14 +454,16 @@ bool BoundingBox<T, NDIMS>::contains(const BoundingBox<OtherT, NDIMS>& otherBB)
 
 //------------------------------------------------------------------------------
 template <typename T, int NDIMS>
-template <typename OtherType>
+template <typename OtherType, int OtherDims>
 AXOM_HOST_DEVICE bool BoundingBox<T, NDIMS>::intersectsWith(
-  const BoundingBox<OtherType, NDIMS>& otherBB) const
+  const BoundingBox<OtherType, OtherDims>& otherBB) const
 {
   bool status = true;
 
   // AABBs cannot intersect if they are separated along any dimension
-  for(int i = 0; i < NDIMS; ++i)
+  constexpr int MinDims = NDIMS < OtherDims ? NDIMS : OtherDims;
```

Comment:
Minor: Could this use `axom::utilities::min` if it were marked as `constexpr`?
(You don't have to make that change, just thinking out loud).

https://github.com/LLNL/axom/blob/c36a089e8fd0672c43e95ca9e22bb93708132032/src/axom/core/utilities/Utilities.hpp#L85-L96

Reply:
I attempted to do that, but I think there was weirdness dealing with references. I may have been doing it wrong - I don't have much experience with writing constexpr functions.

Reply:
Agreed -- I played with it a bit, and it looks like our `min` might have to have value semantics or the input needs to be static to work with `constexpr`.

See: https://godbolt.org/z/n6voz6oKP
(Note: I used `consteval` and c++20 to force compile time evaluation)

Reply:
I removed this code, so we don't need it in this merge request.